test/scalarstats.jl

```diff
@@ -109,8 +109,21 @@ z2 = [8. 2. 3. 1.; 24. 10. -1. -1.; 20. 12. 1. -2.]
 
 @test sem([1:5;]) ≈ 0.707106781186548
 @test sem(skipmissing([missing; 1:5; missing])) ≈ 0.707106781186548
-@test sem(Int[]) === NaN
-@test sem(skipmissing(Union{Int,Missing}[missing, missing])) === NaN
+@test sem(skipmissing([missing; 1:5; missing]), mean=3.0) ≈ 0.707106781186548
+@test sem([1:5;], UnitWeights{Int}(5)) ≈ 0.707106781186548
+@test sem([1:5;], UnitWeights{Int}(5); mean=mean(1:5)) ≈ 0.707106781186548
+@test sem([1:5;], ProbabilityWeights([1:5;])) ≈ 0.6166 rtol=.001
+μ = mean(1:5, ProbabilityWeights([1:5;]))
+@test sem([1:5;], ProbabilityWeights([1:5;]); mean=μ) ≈ 0.6166 rtol=.001
+@test sem([10; 1:5;], ProbabilityWeights([0; 1:5;]); mean=μ) ≈ 0.6166 rtol=.001
+x = sort!(vcat([5:-1:i for i in 1:5]...))
+μ = mean(x)
+@test sem([1:5;], FrequencyWeights([1:5;])) ≈ sem(x)
+@test sem([1:5;], FrequencyWeights([1:5;]); mean=μ) ≈ sem(x)
+@test isnan(sem(Int[]))
+@test isnan(sem(Int[], FrequencyWeights(Int[])))
+@test isnan(sem(Int[], ProbabilityWeights(Int[])))
+@test isnan(sem(skipmissing(Union{Int,Missing}[missing, missing])))
```

Comment:
Can you add `@inferred` to ensure that the implementation is type stable? I.e., use `@test @inferred(sem(....)) \isapprox ...`. It would be good to also test combinations of different types, e.g., `mean::Float64` and `x::Array{Float32}` etc.

Reply:
It looks like the `ProbabilityWeights` code has a type instability and I have no idea why. It works when I remove the block with `isempty`, which is bizarre given it doesn't cause a type instability anywhere else.